Source/Tests/MakeShapeTester.cpp

```diff
@@ -0,0 +1,20 @@
+#include <catch2/catch.hpp>
+#include "Ashkal/Ashkal/Matrix.hpp"
+#include "Ashkal/Ashkal/Point.hpp"
+#include "Ashkal/Shapes/Line.hpp"
+#include "Tests/TestUtility.hpp"
+
+using namespace Ashkal;
+
+TEST_CASE("Testing Creation of Line from Points", "[make_line]") {
+  auto p1 = Point{-2, 2};
+  auto p2 = Point{2, -2};
+  auto line = make_line(p1, p2);
+  auto slope = (p1.y - p2.y) / (p1.x - p2.x);
+  auto angle = (std::atan(1) * 4) - std::abs(atan(slope) - atan(1));
+  auto result = Matrix(
+    {std::cos(angle) * std::sqrt(32), std::sqrt(32), -2},
+    {-std::sqrt(32), std::cos(angle) * std::sqrt(32), 2},
+    {0, 0, 1});
+  REQUIRE(equals(line.get_transformation(), result));
+}
```

Comment:
This test assumes that your implementation is correct which is not a valid way to go about testing things. It's very challenging to come up with a proper test case that is not biased by your implementation but we have to keep the two of them distinct.

In this case, what can we say should be true about our transformation? Well if transformation T takes a line L1 = (-0.5, -0.5) -> (0.5, 0.5) and transforms it into line L2 = p1 -> p2, then T^-1 should take the point p1 and transform it into (-0.5, -0.5) and take the point p2 and transform it into point (0.5, 0.5).

Note that this fact has nothing to do with our implementation and doesn't require us to come up with magic numbers or calculate any kind of cosines or sqrts. It just creates two points, p1, p2... and checks that the inverse transformation takes p1 to (-0.5, -0.5) and takes p2 to (0.5, 0.5).

Reply:
So the inverse transformation of the created line should be equal the transformation of the default line?